spec/httpserver/handlers/handler_spec.clj

```diff
@@ -1,13 +1,27 @@
 (ns httpserver.handler_spec
    (:require [speclj.core :refer :all]
-   [httpserver.handlers.handler :as handler]))
+   [httpserver.handlers.handler :as handler]
+   [httpserver.handlers.teapot :as teapot]
+   [httpserver.routing.router :as router]
+   ))
 
 (import '(com.td.HttpServer HttpResponse))
 (import '(com.td.HttpServer HttpRequest))
 
-(describe "main handler"
+(describe "generating a response"
   (it "returns a HttpResponse with code 418 when passed a HttpRequest GET /coffee"
-    (let [httpRequest (HttpRequest. "GET /coffee HTTP/1.1" (java.util.HashMap. {"this" "that"}))
-          main-handler (handler/new-handler)
-          httpResponse (.generateResponse main-handler httpRequest)]
+      (let [test-list (list {:method "GET" :path "/coffee" :handler teapot/call}
```

Comment:
Rather than `(list {} {} {})`, you could use `'({} {} {})` (a true list) or `[{} {} {}]` (a vector). You'll probably see the vector more often—lists are most often employed when working with macros. See [this](http://stackoverflow.com/questions/11504236/as-a-data-container-what-are-the-main-differences-between-vector-and-list) and [this](http://stackoverflow.com/questions/1147975/in-clojure-when-should-i-use-a-vector-over-a-list-and-the-other-way-around) for descriptions of the differences between lists and vectors.

Also, just FYI, the quote in the example above is literally called "quote", and is not a typo. As Colin describes in his Clojure Macros book, you can think of it as meaning "literally". So, `(a b c)` = "call a with b and c as args", while `'(a b c)` = "literally, a b c (in a list)"

This is in a few places throughout the code. I'd recommend converting all `(list a b c)` 's to `[a b c]`